### PR TITLE
Vulkan support

### PR DESCRIPTION
**Pull request by guptaNswati**
@RenaudWasTaken PTAL

### CODE REVIEW

**Comment by guptaNswati** [comment]
@rajatchopra PTAL

File: src/nvc_mount.c [hunk 1]
```diff
@@ -404,6 +404,10 @@ symlink_libraries(struct error *err, const struct nvc_container *cnt, const char
                         /* XXX GLVND requires this symlink for indirect GLX support. */
                         if (symlink_library(err, paths[i], lib, "libGLX_indirect.so.0", cnt->uid, cnt->gid) < 0)
                                 return (-1);
+                } else if (str_has_prefix(lib, "libnvidia-opticalflow.so")) {
+                        /* XXX Fix missing symlink for libnvidia-opticalflow.so. */
+                        if (symlink_library(err, paths[i], "libnvidia-opticalflow.so.1", "libnvidia-opticalflow.so", cnt->uid, cnt->gid) < 0)
```
Comment (rajatchopra) [inline src/nvc_mount.c]: Don't these symlinks get fixed by ldconfig? We can just mount the libraries themselves only.

**Comment by RenaudWasTaken** [comment]
Ideally these should be 3 separate commits

Can you detail a bit what actually needs to be done to enable Vulkan support? I can't really assess the change if I don't know how it works 😛

File: src/nvc_mount.c (hunk 1 above)
Comment (guptaNswati) [reply src/nvc_mount.c]: This is to fix an internal bug. ldconfig links with libnvidia-opticalflow.so.1 but some optics applications expects libs to be libnvidia-opticalflow.so similar to libcuda.so.

**Comment by guptaNswati** [comment]
@RenaudWasTaken For running vulkan application inside the container, libnvidia-glvkspirv.so.$DRIVER_VERSION is the host lib that needs to be mounted inside the container, so would be better to add it from libnvidia. Attaching the strace of the libs vulkan application tries to open. 
[swati.txt](https://github.com/NVIDIA/libnvidia-container/files/3412452/swati.txt)

Similar issue https://github.com/NVIDIA/libnvidia-container/issues/55

For more information, please take a look at the Dockerfile for Vulkan in the internal repo that will help shed light on other libs needed for Vulkan.

**Comment by RenaudWasTaken** [comment]
lgtm
Move this PR to gitlab :)

**Pull request closed without merging**